openpype/plugins/publish/extract_color_transcode.py

```diff
@@ -0,0 +1,333 @@
+                    files_to_convert = [new_repre["files"]]
+
+                output_extension = output_def["extension"]
+                output_extension = output_extension.replace('.', '')
+                self._rename_in_representation(new_repre,
+                                               files_to_convert,
+                                               output_name,
+                                               output_extension)
+
+                target_colorspace = output_def["colorspace"]
+                view = output_def["view"] or colorspace_data.get("view")
+                display = (output_def["display"] or
+                           colorspace_data.get("display"))
+                # both could be already collected by DCC,
+                # but could be overwritten
+                if view:
+                    new_repre["colorspaceData"]["view"] = view
+                if display:
+                    new_repre["colorspaceData"]["display"] = display
+
+                files_to_convert = self._translate_to_sequence(
+                    files_to_convert)
+                for file_name in files_to_convert:
+                    input_path = os.path.join(original_staging_dir,
+                                              file_name)
+                    output_path = self._get_output_file_path(input_path,
+                                                             new_staging_dir,
+                                                             output_extension)
+                    convert_colorspace(
+                        input_path,
+                        output_path,
+                        config_path,
+                        source_colorspace,
+                        target_colorspace,
+                        view,
+                        display,
+                        self.log
+                    )
+
+                # cleanup temporary transcoded files
+                for file_name in new_repre["files"]:
+                    transcoded_file_path = os.path.join(new_staging_dir,
+                                                        file_name)
+                    instance.context.data["cleanupFullPaths"].append(
+                        transcoded_file_path)
+
+                custom_tags = output_def.get("custom_tags")
+                if custom_tags:
+                    if not new_repre.get("custom_tags"):
+                        new_repre["custom_tags"] = []
+                    new_repre["custom_tags"].extend(custom_tags)
+
+                # Add additional tags from output definition to representation
+                if not new_repre.get("tags"):
+                    new_repre["tags"] = []
+                for tag in output_def["tags"]:
+                    if tag not in new_repre["tags"]:
+                        new_repre["tags"].append(tag)
+
+                    if tag == "review":
+                        added_review = True
+
+                new_repre["tags"].append("newly_added")
```

Comment:
What about create separated list of "new representations" to avoid using the tag?
```
new_representations = []
for repre in repres:
    ...
    new_representations.append(new_repre)
    
for repre in tuple(repres):
    if <remove condition>:
        repres.remove(repre)

repres.extend(new_representations)
```

Reply:
That is actually much nicer approach than what I concocted.